include/sparrow/ordering.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <compare>
+#include <concepts>
+
+namespace sparrow
+{
+    template <typename T1, typename T2>
+    concept StrongOrderable = requires(T1 a, T2 b) {
+        { a.value() <=> b.value() } -> std::same_as<std::strong_ordering>;
+    };
```

Comment:
How is `StrongOrderable` be picked over `PartialOrderable` when condition on `T1` and `T2` are met?

Reply:
It's about the returned type pf the spaceship operator